receiver/signalfxreceiver/receiver.go

```diff
@@ -165,28 +175,21 @@ func (r *sfxReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *sfxReceiver) handleReq(resp http.ResponseWriter, req *http.Request) {
-	transport := "http"
-	if r.config.TLSSetting != nil {
-		transport = "https"
-	}
-	ctx := obsreport.ReceiverContext(req.Context(), r.config.Name(), transport, r.config.Name())
-	ctx = obsreport.StartMetricsReceiveOp(ctx, r.config.Name(), transport)
-
+func (r *sfxReceiver) preRequest(ctx context.Context, resp http.ResponseWriter, req *http.Request) ([]byte, bool) {
```

Comment:
Please add a comment to explain what this func does and what the return values are. It is not clear from the signature. Perhaps a better func name would help.

Reply:
Renamed to `readBody`.